code/go/0chain.net/blobbercore/allocation/renamefilechange.go

```diff
@@ -41,11 +41,10 @@ func (rf *RenameFileChange) ApplyChange(ctx context.Context, change *AllocationC
 		return nil, common.NewError("invalid_reference_path", "file already exists")
 	}
 
-	affectedRef, err := reference.GetObjectTree(ctx, rf.AllocationID, rf.Path)
+	affectedRef, err := rootRef.GetSrcPath(rf.Path)
```

Comment:
This is not correct. You need to use GetObjectTree.

Reply:
SrcPath function returns the same thing without making calls to DB, there could be a previous change which affected the children of srcRef and that change would have beeb saved in memory.

Reply:
SrcPath is taken from reference path. It does not contain whole files that are affected by rename operation.

Reply:
this object tree should have been already loaded before, we can pass reference to it, or convert it to scrPath